test/models/foreman_dlm/dlmlock_test.rb

```diff
@@ -208,5 +202,45 @@ def callback2
         assert_equal Dlmlock::Update.find(dlmlock.id), Dlmlock.search_for("name ~ #{dlmlock.name}").first
       end
     end
+
+    context '#log_events' do
+      let(:dlmlock) { FactoryBot.create(:dlmlock) }
+
+      test 'logs acquire event' do
+        assert_difference  -> { DlmlockEvent.count }, 1 do
+          dlmlock.acquire!(host1)
+        end
+      end
+
+      test 'logs disable event' do
+        assert_difference  -> { DlmlockEvent.count }, 1 do
+          dlmlock.disable!
+        end
+      end
+
+      test 'logs enable event' do
+        dlmlock.update(enabled: false)
+
+        assert_difference  -> { DlmlockEvent.count }, 1 do
+          dlmlock.enable!
+        end
+      end
+
+      test 'logs failed event' do
+        dlmlock.acquire!(host1)
+
+        assert_difference  -> { DlmlockEvent.count }, 1 do
+          dlmlock.disable!
```

Comment:
This should be `dlmlock.acquire!(host2)`.

Can you also check if events for the correct hosts and type were created?